File: testSuite/test-bar-instability.pl
#!/usr/bin/env perl
use strict;
use warnings;
use PDL;
use PDL::NiceSlice;
use PDL::IO::HDF5;

# Run models that test that the bar instability mechanism is able to create spheroids.
# Andrew Benson (25-July-2018)

# Make output directory.
system("mkdir -p outputs/");

# Run the model.
system("cd ..; ./Galacticus.exe testSuite/parameters/barInstability.xml");
if ( $? == 0 ) {
    my $model               = new PDL::IO::HDF5("outputs/barInstability.hdf5");
    my $outputs             = $model  ->group  ('Outputs'            )       ;
    my $output              = $outputs->group  ('Output1'            )       ;
    my $nodes               = $output ->group  ('nodeData'           )       ;
    my $spheroidMassStellar = $nodes  ->dataset('spheroidMassStellar')->get();
    if ( $spheroidMassStellar->((0)) <= 0.0 ) {
	print "FAIL: spheroid has non-positive mass\n";
    } else {
	print "SUCCESS\n";
    }
} else {
    print "FAIL: bar instability test model failed to run\n";
}

exit;

File: testSuite/test-bar-instability.py
#!/usr/bin/env python3
import subprocess
import sys
import h5py
import numpy as np

# Test bar instability model.
# Andrew Benson (ported to Python)

# Ensure output directory exists.
subprocess.run("mkdir -p outputs", shell=True)

# Run the model.
status = subprocess.run("cd ..; ./Galacticus.exe testSuite/parameters/barInstability.xml", shell=True)
if status.returncode != 0:
    print("FAILED: model failed to run")
    sys.exit(0)

# Read model data.
with h5py.File("outputs/barInstability.hdf5", "r") as model:
    spheroidMassStellar = model["Outputs/Output1/nodeData/spheroidMassStellar"][:]

if spheroidMassStellar[0] > 0.0:
    print("SUCCESS: bar instability")
else:
    print("FAILED: bar instability - no stellar mass in spheroid")
    sys.exit(0)
